spec/pokedex_spec.rb

```diff
@@ -0,0 +1,349 @@
+                stat: {
+                    name: "Defense"
+                }
+            },
+            {
+                base_stat: 65,
+                stat: {
+                    name: "Special-Attack"
+                }
+            },
+            {
+                base_stat: 65,
+                stat: {
+                    name: "Special-Defense"
+                }
+            },
+            {
+                base_stat: 45,
+                stat: {
+                    name: "Speed"
+                }
+            }
+        ],
+        types: [
+            {
+                slot: 1,
+                type: {
+                    name: "Grass"
+                }
+            },
+            {
+                slot: 2,
+                type: {
+                    name: "Poison"
+                }
+            }
+        ],
+        weight: 69
+    })
+    
+    pokedex.push({
+        id: 4,
+        name: "Charmander",
+        stats: [
+            {
+                base_stat: 39,
+                stat: {
+                    name: "HP"
+                }
+            },
+            {
+                base_stat: 52,
+                stat: {
+                    name: "Attack"
+                }
+            },
+            {
+                base_stat: 43,
+                stat: {
+                    name: "Defense"
+                }
+            },
+            {
+                base_stat: 60,
+                stat: {
+                    name: "Special-Attack"
+                }
+            },
+            {
+                base_stat: 50,
+                stat: {
+                    name: "Special-Defense"
+                }
+            },
+            {
+                base_stat: 65,
+                stat: {
+                    name: "Speed"
+                }
+            }
+        ],
+        types: [
+            {
+                slot: 1,
+                type: {
+                    name: "Fire"
+                }
+            },
+            {
+                slot: 2,
+                type: {
+                    name: "N/A"
+                }
+            }
+        ],
+        weight: 85
+    })
+  end
+
+describe 'Pokedex' do
+
+  context 'The correct regions are selected based on the methods parameter' do
+
+      it 'sets lower_limit to 1 and upper_limit to 151 when the region is kanto' do
+          testdex = Pokedex.new("", "kanto")
+
+          expect(testdex.lower_limit).to eq(1)
+          expect(testdex.upper_limit).to eq(151)
+      end
+
+      it 'sets lower_limit to 152 and upper_limit to 252 when the region is johto' do
+          testdex = Pokedex.new("", "johto")
+
+          expect(testdex.lower_limit).to eq(152)
+          expect(testdex.upper_limit).to eq(251)
+      end
+  end
+
+  context 'The data retrieved from the API has been reformated correctly' do
+    let(:one_type_pokemon) {
+        one_type_pokemon = {
+            "base_experience": 62,
+            "forms": [
```

Comment:
indenting looks odd here

Reply:
It was off but a tab or two in places due to me not adjusting it after pasting the data in. I have now fixed the indentation